src/main/java/au/org/ala/biocache/service/DownloadService.java

```diff
@@ -1330,7 +1328,7 @@ public DownloadDetailsDTO call() throws Exception {
                                     .replace("[hubName]",hubName),
                                     null);
 
-                            if (currentDownload != null && currentDownload.getFileLocation() != null) {
+                            if (currentDownload.isEmailNotify() && currentDownload != null && currentDownload.getFileLocation() != null) {
```

Comment:
The non-email actions in this block need to have a chance to still run. One example is the DOI minting that should still proceed if they specified `mintDoi` in their query. The DOI minting code could either be moved outside of this block, or `if (currentDownloads.isEmailNotify())` could be moved further down in the method.

Reply:
Also, if currentDownload is null for whatever reason, this would throw a `NullPointerException` as it is before the null-guard.

Reply:
I've moved the check to after the DOI minting process.